### ISSUE TITLE
🐛 Json schema validation fails when for `_ab_cdc_deleted_at` or `_ab_cdc_updated_at` is null

### ISSUE THREAD

**Issue by smartpierre** [airbytehq/airbyte#12804]
## Environment
- **Airbyte version**:  0.38.1-alpha
- **OS Version / Instance**: AWS EC2
- **Deployment**: Docker
- **Source Connector and version**: Postgres 0.4.4 and Mysql 0.5.4
- **Destination Connector and version**: Snowflak 0.4.25
- **Severity**: Medium
- **Step where error happened**: Sync job

## Current Behavior
We upgraded our airbyte version from 0.35.68-alpha to 0.38.1-alpha since then, we are having issues when running sync jobs. The log file are filled with json schema validation errors (one, for each database record, see below). At some point, it filled our server storage and Airbyte was unresponsive. I had to delete log files, one was 25Gb (for a 9Gb database sync).
Note that **these errors don't make the sync job fails directly**, they just fill the storage until the server itself fails entirely.
Also note, that we have the same issue on the Postgres and the MySQL source connectors.
Another note, it looks like we only have this issue where CDC is enabled.

I'm willing to send a complete log file and show our `.env` to an Airbyte employee.

## Expected Behavior
I just want it to be like it was before :(

## Logs

<details>
<summary>LOG</summary> 

```

2022-05-11 15:04:34 WARN i.a.w.DefaultReplicationWorker(lambda$getReplicationRunnable$5):298 - Record schema validation failed. Errors: json schema validation failed when comparing the data to the json schema. 
Errors: $._ab_cdc_deleted_at: null found, string expected 
Schema: 
{
  "type" : "object",
  "properties" : {
    "user" : {
      "type" : "string"
    },
    "device" : {
      "type" : "string"
    },
    "record" : {
      "type" : "string"
    },
    "_ab_cdc_lsn" : {
      "type" : "number"
    },
    "_ab_cdc_deleted_at" : {
      "type" : "string"
    },
    "_ab_cdc_updated_at" : {
      "type" : "string"
    }
  },
  "$schema" : "http://json-schema.org/draft-07/schema#"
}

```

</details>

## Steps to Reproduce
1. Upgrade Airbyte from 0.35.68-alpha to 0.38.1-alpha
2. Run sync that were working
3. Watch the log

## Are you willing to submit a PR?

Remove this with your answer.

Anyway, if somebody has any clue, what I could try to fix it, I'm all ears.

Thanks in advance 🙏

**Comment by damiantw** [comment]
I'm experiencing the same issue with our Postgres <> Redshift connections on 0.38.2-alpha. It seems to write a log entry for every record processed in a table that has nullable columns. 

I'm going to try raising the log level to ERROR for the time being.

**Comment by damiantw** [comment]
Raising the worker log level to ERROR will prevent the disk from filling up. However, my sync jobs are running noticeably slower than usual. It's hard to tell if the validation failure is the culprit since I'm not getting progress (INFO) output from the source or destination containers. 

The schema validation seems to have been introduced in [this commit](https://github.com/airbytehq/airbyte/commit/02dec04b74d7b08791be3d43ee701bb32b72ec92). 

The JSON schemas in my Postgres source catalogs do not include null types for nullable columns. It seems that the reflection logic for building JSON schemas for certain sources needs to be updated to account for nullable fields or the logging logic needs to be adjusted to disregard null type validation errors.

**Comment by RobertoBonnet** [comment]
Same here
![image](https://user-images.githubusercontent.com/19189101/168168850-7441d584-9039-4684-956d-1c3ce23c5dc7.png)

**Comment by clovismasson** [comment]
Hi, same issue for me here.

Since almost two days, none of my postgresql source jobs worked, without had making any modification or upgrade (currently `v.0.35.67`). I tried to update my source and destination (`psql` <-> `BQ`) to their latest versions but it did not fix anything.

Here are an logfile example on one of my latest runs:
[logs-286.txt](https://github.com/airbytehq/airbyte/files/8684650/logs-286.txt)

**Comment by PierreKerschgens** [comment]
Hey everybody,

I'm facing the same issue for my PostgreSQL jobs after upgrading Airbyte from 0.35.6-alpha to 0.38.1-alpha. After noticing this issue I tried to downgrade the postgres source from 0.4.14 to 0.4.2 (version before Airbyte upgrade) but this broke my whole postgres setup. In the end I had to restore pre-upgrade snapshots of my Airbyte setup.

**Comment by ciminelli** [comment]
Same issue happening for me as well and nothing has been changed, just started failing yesterday. Looks like it's failing on the BigQuery upload, wonder if something changed on their end.

```
2022-05-13 11:51:48 [43mdestination[0m >  details: 400 Bad Request
2022-05-13 11:51:48 [43mdestination[0m > PUT https://www.googleapis.com/upload/bigquery/v2/projects/xxxxxxxxxxx
2022-05-13 11:51:48 [43mdestination[0m > {
2022-05-13 11:51:48 [32mINFO[m i.a.v.j.JsonSchemaValidator(test):56 - JSON schema validation failed. 
errors: $: string found, object expected
2022-05-13 11:51:48 [1;31mERROR[m i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$1):70 - Validation failed: "error"
2022-05-13 11:51:48 [32mINFO[m i.a.v.j.JsonSchemaValidator(test):56 - JSON schema validation failed. 
errors: $: string found, object expected
2022-05-13 11:51:48 [1;31mERROR[m i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$1):70 - Validation failed: "code"
2022-05-13 11:51:48 [32mINFO[m i.a.v.j.JsonSchemaValidator(test):56 - JSON schema validation failed. 
errors: $: string found, object expected
2022-05-13 11:51:48 [1;31mERROR[m i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$1):70 - Validation failed: "message"
2022-05-13 11:51:48 [32mINFO[m i.a.v.j.JsonSchemaValidator(test):56 - JSON schema validation failed. 
errors: $: string found, object expected
2022-05-13 11:51:48 [1;31mERROR[m i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$1):70 - Validation failed: "errors"
2022-05-13 11:51:48 [43mdestination[0m >       {
2022-05-13 11:51:48 [32mINFO[m i.a.v.j.JsonSchemaValidator(test):56 - JSON schema validation failed. 
errors: $: string found, object expected
2022-05-13 11:51:48 [1;31mERROR[m i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$1):70 - Validation failed: "message"
2022-05-13 11:51:48 [32mINFO[m i.a.v.j.JsonSchemaValidator(test):56 - JSON schema validation failed. 
errors: $: string found, object expected
2022-05-13 11:51:48 [1;31mERROR[m i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$1):70 - Validation failed: "domain"
2022-05-13 11:51:48 [32mINFO[m i.a.v.j.JsonSchemaValidator(test):56 - JSON schema validation failed. 
errors: $: string found, object expected
2022-05-13 11:51:48 [1;31mERROR[m i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$1):70 - Validation failed: "reason"
2022-05-13 11:51:48 [43mdestination[0m >       }
2022-05-13 11:51:48 [43mdestination[0m >     ],
2022-05-13 11:51:48 [32mINFO[m i.a.v.j.JsonSchemaValidator(test):56 - JSON schema validation failed. 
errors: $: string found, object expected
2022-05-13 11:51:48 [1;31mERROR[m i.a.w.p.a.DefaultAirbyteStreamFactory(lambda$create$1):70 - Validation failed: "status"
2022-05-13 11:51:48 [43mdestination[0m >   }
2022-05-13 11:51:48 [43mdestination[0m > }
```

**Comment by nscheifler** [comment]
Following - I am having the same issues as the rest of this group - running version `0.35.12` and did not upgrade - suddenly all my postgresql source jobs failing with 400 Bad Request and JSON schema validation errors

**Comment by marcosmarxm** [comment]
> Following - I am having the same issues as the rest of this group - running version `0.35.12` and did not upgrade - suddenly all my postgresql source jobs failing with 400 Bad Request and JSON schema validation errors

This issue was related to Bigquery https://github.com/airbytehq/airbyte/issues/12813

**Comment by alovew** [comment]
This issue is resolved in v0.38.4-alpha. Schema validation errors are now consolidated before being logged, so there will be a maximum of one log line per stream rather than a log line for each record.

**Comment by smartpierre** [comment]
Hello, I did the update, and it does look like this issue is fixed on my side. 

Thank you very much @alovew for your quick fix :)
